src/layers.cpp

```diff
@@ -12,27 +12,32 @@ static void handleKeymapKeyswitchEvent(Key keymapEntry, uint8_t keyState) {
   if (keymapEntry.keyCode >= MOMENTARY_OFFSET) {
     uint8_t target = keymapEntry.keyCode - MOMENTARY_OFFSET;
 
-    if (keyToggledOn(keyState)) {
-      if (target == KEYMAP_NEXT) {
+    switch (target) {
+    case KEYMAP_NEXT:
+      if (keyToggledOn(keyState))
         Layer.next();
-      } else if (target == KEYMAP_PREVIOUS) {
+      else if (keyToggledOff(keyState))
         Layer.previous();
-      } else {
-        Layer.on(target);
-      }
-    }
-    if (keyToggledOff(keyState)) {
-      if (target == KEYMAP_NEXT) {
+      break;
+
+    case KEYMAP_PREVIOUS:
+      if (keyToggledOn(keyState))
         Layer.previous();
-      } else if (target == KEYMAP_PREVIOUS) {
+      else if (keyToggledOff(keyState))
         Layer.next();
-      } else {
+      break;
+
+    default:
+      if (keyIsPressed(keyState)) {
```

Comment:
Can you add some comments to the switch statement? Primarily, the "why" behind the logic in the 'default' case. It's not something that's intuitively obvious from reading the code